lib/mongoid/extensions/big_decimal.rb

```diff
@@ -58,7 +68,17 @@ def demongoize(object)
         #
         # @return [ String, nil ] A String representing the object or nil.
         def mongoize(object)
-          object && object.numeric? ? object.to_s : nil
+          if object
+            if Mongoid.map_big_decimal_to_decimal128
+              if object.is_a?(BigDecimal)
+                BSON::Decimal128.new(object)
+              elsif object.numeric?
+                BSON::Decimal128.new(object.to_s)
+              end
+            elsif object.numeric?
+              object.to_s
+            end
```

Comment:
The outer `if` should also check for BSON::Decimal128 and have the `else` branch, same as the inner `if`.

Reply:
done